session/pingpong/exchange_message_tracker.go

```diff
@@ -27,11 +33,35 @@ import (
 	"github.com/pkg/errors"
 )
 
+// ErrWrongProvider represents an issue where the wrong provider is supplied
```

Comment:
Minor thing. I suggest to end each comment with a dot to end sentence because you can have more than one sentence it comment block. 

Example from Godoc https://blog.golang.org/godoc-documenting-go-code
Example from go std https://github.com/golang/go/blob/master/src/bytes/bytes.go#L439